### DISCUSSION TITLE
The most common layout

### DISCUSSION THREAD

**Discussion by mturoci** [h2oai/wave#1074] [Show and tell]
## Answer

```py
from h2o_wave import Q, ui, main, app


@app('/demo')
async def serve(q: Q):
    q.page['meta'] = ui.meta_card(box='', layouts=[ui.layout(breakpoint='xs', zones=[
        ui.zone(name='main', size='100vh', zones=[
            ui.zone(name='header', size='80px'),
            ui.zone(name='body', size='1'),
            ui.zone(name='footer', size='60px'),
        ]),
    ])])

    q.page['header'] = ui.header_card(box='header', title='Header', subtitle='Subtitle')
    q.page['body'] = ui.form_card(box='body', items=[ui.text(content='I am body!')])
    q.page['footer'] = ui.footer_card(box='footer', caption='Footer caption')
    await q.page.save()
```

## Explanation

The general idea is to predefine our app wrapper height in advance (`100vh`), render known height elements (header/footer) with fixed sizes and let our body fill the rest of the available space (`size=1`).

Note: `100vh` means 100% of viewport height - visible screen size.

## Result

![image](https://user-images.githubusercontent.com/64769322/137846683-be8595f0-f6b2-4643-8313-c8d0e4eb5631.png)

https://user-images.githubusercontent.com/64769322/137846783-51fdd368-a7c3-453b-afbb-6035fb2c74b7.mov

**Comment by mtanco** [comment]
🎉 awesome! @mturoci Can we get this in the IDE hints?!

**Reply by mturoci** [reply]
@mtanco sure, but we need to release them first :)
